src/Components/Components/src/DefaultComponentActivator.cs

```diff
@@ -10,10 +11,11 @@ internal class DefaultComponentActivator : IComponentActivator
         public static IComponentActivator Instance { get; } = new DefaultComponentActivator();
 
         /// <inheritdoc />
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2067", Justification = "Requires a gesture that ensures components are always preserved. https://github.com/mono/linker/issues/1806")]
         public IComponent CreateInstance(Type componentType)
         {
             var instance = Activator.CreateInstance(componentType);
-            if (!(instance is IComponent component))
+            if (instance is not IComponent component)
```

Comment:
This should use Type.IsAssignableFrom to check whether componentType implements the interface **before** activating the type. Otherwise this suppression will be invalid even if illink adds support for the gesture that ensures components are always preserved. If the type is not a component, activator might throw a MissingMethodException because trimming removed the ctor before we reach this check to throw ArgumentException.

The goal of trimming annotations is to never have differences is behaviors at runtime caused by trimming.